packages/docs/docs/getting-started/create-unirep-app.md

```diff
@@ -312,8 +313,36 @@ function submitAttestation(
 }
 ```
 
+**e.g.**<br/>
+An attester needs to track of the total number of posts a user has made. The code below executes an attestation to increment the user's post count when the user submits a valid post:
+
+```js
+<input 
+  type='submit'
+  value='POST'
+  onClick={async () => {
+    // check if user is transitioned to current epoch
+    if (
+      user.userState &&
+      user.userState.sync.calcCurrentEpoch() !==
+        (await user.userState.latestTransitionedEpoch())
+    ) {
+        throw new Error('Needs transition')
+    }
+    // create new entry in off-chain database
+    await app.createNewPost(postData)
+    // submit attestation to increment user's post count
+    await user.requestReputation(
+      {[0]: 1},
+      reqInfo.nonce ?? 0
+    )
+  }}
+/>
+```
+
 :::info
-See [`attesterCurrentEpoch`](../contracts-api/unirep-sol.md#attestercurrentepoch), [`attest`](../contracts-api/unirep-sol.md#attest)
+See [`attesterCurrentEpoch`](../contracts-api/unirep-sol.md#attestercurrentepoch)<br/>
+See [`attest`](../contracts-api/unirep-sol.md#attest)
 :::
 
 ### Verify epoch key on-chain
```

Comment:
@vivianjeng i haven't made many changes to this section, i'm having a hard time understanding the purpose of the example. i thought that an epoch key proof is made when an attestation is initiated, and verified before the data gets changed. is that correct or no? in the example here it looks like you are modifying the contract to do that, so i am confused.

additionally, this section feels a bit out of place.  probably because i don't understand it 😅 - but the rest of the docs feels like an introduction to concepts, getting ideas about user flow.. and then this section feels like it might belong in a different place with other docs that walk through function implementation.  ??

Reply:
> i thought that an epoch key proof is made when an attestation is initiated, and verified before the data gets changed.

yeah
and the epoch key proof can be verified either off-chain (by relayer) or on-chain (by the smart contract like this)
so on-chain verification also works in this example.

I want to let developers have some idea to modify the smart contract
and maybe provide ideas of some useful functions (e.g. `verifyEpochKeyProof`, `decodeEpochKeySignals`,...)
or it might be hard to find what they need in https://developer.unirep.io/docs/contracts-api/unirep-sol

Reply:
ok i understand thanks! i will add a bit more explanation here.

Reply:
@vivianjeng this is ready to merge if you are ok with my changes.